aiohttp/web_urldispatcher.py

```diff
@@ -830,6 +840,10 @@ def register_resource(self, resource):
     def add_resource(self, path, *, name=None):
         if path and not path.startswith('/'):
             raise ValueError("path should be started with / or be empty")
+        # avoid to create unnecessary resources
+        resource = self._named_resources.get(name)
```

Comment:
The only last resource in table should be checked -- otherwise the PR changes route order implicitly.